File: src/main/java/com/github/kayjamlang/core/exceptions/ParserException.java
package com.github.kayjamlang.core.exceptions;

import com.github.kayjamlang.core.KayJamLexer;

public class ParserException extends Exception {
    public ParserException(KayJamLexer lexer, String message){
        super(message+" in "+lexer.getLine()+" line");
    }

    public ParserException(int line, String message){
        super(message+" in "+line+" line");
    }
}

File: src/main/scala/com/github/kayjamlang/core/exception/ParserException.scala
package com.github.kayjamlang.core.exception

import com.github.kayjamlang.core.KayJamLexer

class ParserException(message: String) extends Exception(message) {
  def this(lexer: KayJamLexer, message: String) {
    this(s"$message in ${lexer.getLine} line")
  }

  def this(line: Int, message: String) {
    this(s"$message + in + $line + line")
  }
}